### PR TITLE
Add endpoint description

### PR DESCRIPTION
**Pull request by howjmay**

### CODE REVIEW

**Comment by jkrvivian** [comment]
@HowJMay  need rebase, thanks!

File: paper.tex [hunk 1]
```diff
@@ -291,17 +291,43 @@ \subsubsection{A Scalable Keys and Data Entry Points Management}
...
 \label{section:ta_endpoint}
-MAM builds a secure and authenticated communication protocol on top of multiple cyptosystems, which low-level devices need to spend a lot of resources to apply it, and they may not even support built-in libraries in regular operating system. To overcome these concerns, we propose another approach that allows data providers to delegate the MAM operations to Tangle-accelerators, proxy servers with high computation power, which can accelerates the interactions with Tangle while ensuring the privacy of data providers.
+MAM builds a secure and authenticated communication protocol on top of multiple cryptosystems, which low-level devices need to spend a lot of resources to apply it, and they may not even support built-in libraries in regular operating system. To overcome these concerns, we propose another approach that allows data providers to delegate the MAM operations to Tangle-accelerators, proxy servers with high computation power, which can accelerates the interactions with Tangle while ensuring the privacy of data providers.
 
-\subsubsection{Plight for Low-level edge device}
+\subsubsection{Plight for Low-level Edge Device}
```
Comment (jkrvivian) [inline paper.tex]: Device`s`

File: paper.tex [hunk 2]
```diff
@@ -291,17 +291,43 @@ \subsubsection{A Scalable Keys and Data Entry Points Management}
...
+MAM requires quite huge computational power which causes a burden for low-level devices to adopt MAM service. Therefore we introduce a proxy server, tangle-accelerator, which performs the heavy MAM computations, so low-level devices can be relieved from the enormous computation.
```
Comment (jkrvivian) [inline paper.tex]: This paragraph is actually similar to the previous one.
Can you make them into one paragraph?
Then maybe we can also remove the subsubsection `Plight...` ... :thinking:

File: paper.tex [hunk 3]
```diff
@@ -291,17 +291,43 @@ \subsubsection{A Scalable Keys and Data Entry Points Management}
...
+
+\subsubsection{Infrastructure}
+To build the system, MQTT is chosen as the communication protocol. MQTT protocol utilizes much more lightweight packet than HTTP protocol has. The topic-based publish-subscribe paradigm and lightweight packet that MQTT adopts would undoubtedly play an important role on low-level edge devices with limited bandwidth. The publish-subscribe model decouples publishers and subscribers in many ways.
```
Comment (jkrvivian) [inline paper.tex]: At the end, change `.` to `:`

File: paper.tex [hunk 4]
```diff
@@ -291,17 +291,43 @@ \subsubsection{A Scalable Keys and Data Entry Points Management}
...
+
+\begin{itemize}
+    \item Space decoupling: Space decoupling allows subscriber and publisher don’t need to know the information of each other
+    \item Time decoupling: Publishers and subscribers do not need to run at the same time
+    \item Synchronization decoupling: Operations are decoupled from the publisher and subscriber. If we can decouple the operations, then the endpoint doesn’t need to maintain the connection until receive the response. Message from endpoint can be served as send-and-forget fashion.
+\end{itemize}
+
+\subsubsection{End-to-End-Encryption}
+Blockchain operations are known for high computational power consuming. Such a characteristics makes a burden for including low-level embedded system to blockchain service. Therefore, we introduce a proxy server which would process the blockchain operations for the low-level embedded system. However, introducing a proxy server to process all the cryptographic operations causes the proxy server or the middle man can easily tamper the message.
```
Comment (jkrvivian) [inline paper.tex]: The sentences before `However,` is similar to the first paragraph.
Remove and we can move on to the concerns directly.

File: paper.tex [hunk 5]
```diff
@@ -291,17 +291,43 @@ \subsubsection{A Scalable Keys and Data Entry Points Management}
...
+
+To avoid such malicious operations happened during transmission, we introduce another lightweight end-to-end-encryption (E2EE) upon MAM protocol. The plaintext will be encrypted with this lightweight end-to-end-encryption. The E2EE process ensures only authorized subscribers can read the messages published on MAM message channel chain.
```
Comment (jkrvivian) [inline paper.tex]: * To avoid such malicious operations ~~happened~~ during transmission
* `on MAM`

File: paper.tex [hunk 6]
```diff
@@ -291,17 +291,43 @@ \subsubsection{A Scalable Keys and Data Entry Points Management}
...
+
+\subsubsection{Register Machine}
+User with channel seed can acquire channel chain ownership. Seed is generated by secret hardware information on edge device, so it should be transmitted to tangle-accelerator which allows tangle-accelerator to publish messages on MAM channel chain. Asymmetric encryption is utilizing during the process of exchanging Seed. Digital signature should be taken at the same time to ensure data integrity. Once seed has been successfully exchanged, edge device is successfully registered on the specific tangle-accelerator. Seed isn't needed to be exposed during communication after an edge device is successfully registered.
```
Comment (jkrvivian) [inline paper.tex]: I think `unique hardware information` is better.

File: paper.tex (hunk 5 above)
Comment (howjmay) [reply paper.tex]: on MAM?

Comment (jkrvivian) [reply paper.tex]: The last sentence. `on MAM channel.`

Comment (howjmay) [reply paper.tex]: We need a term to describe the channels belong to a single seed

Comment (howjmay) [reply paper.tex]: Any idea?

Comment (jkrvivian) [reply paper.tex]: why?
I mentioned this relationship in the previous section `The message string`, people should get the channel is generated from a seed.
But you have to pull `Chap4_mam` first. :smile:

**Comment by howjmay** [comment]
The description for endpoint is done. Now it is lack of experiment in evaluation section only.

File: paper.tex [hunk 7]
```diff
@@ -296,17 +296,57 @@ \subsubsection{A Scalable Keys and Data Entry Points Management}
...
+MAM builds a secure and authenticated communication protocol on top of multiple cryptosystems, which low-level devices need to spend a lot of resources to apply it, and they may not even support built-in libraries in regular operating system. To overcome these concerns, we propose another approach that allows data providers to delegate the MAM operations to Tangle-accelerators, proxy servers with high computation power, which can accelerates the interactions with Tangle while ensuring the privacy of data providers.
 
-\subsubsection{Plight for Low-level edge device}
+\subsubsection{Infrastructure}
+To build the system, MQTT is chosen as the communication protocol. MQTT protocol utilizes much more lightweight packet than HTTP protocol has. The topic-based publish-subscribe paradigm and lightweight packet that MQTT adopts would undoubtedly play an important role on low-level edge devices with limited bandwidth. The publish-subscribe model decouples publishers and subscribers in many ways:
```
Comment (jserv) [inline paper.tex]: Change "publish-subscribe paradigm" to "publish-subscribe pattern"

File: paper.tex [hunk 8]
```diff
@@ -296,17 +296,57 @@ \subsubsection{A Scalable Keys and Data Entry Points Management}
...
+
+\begin{itemize}
+    \item Space decoupling: Space decoupling allows subscriber and publisher don’t need to know the information of each other
```
Comment (jserv) [inline paper.tex]: Fix the grammar.

**Pull request merged**